### PR TITLE
test: Apply flake8 to tests/ directory

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Refactoring

### CODE REVIEW

File: tests/test_domain_c.py [hunk 1]
```diff
@@ -426,7 +426,7 @@ def test_union_definitions():
     check('struct', '{key}A', {1: 'A'})
 
 
-def test_union_definitions():
+def test_union_definitions2():
```
Comment (tk0miya) [inline tests/test_domain_c.py]: @jakobandersen This case overrides the above one. Could you give a good name for this in another PR, please?

File: tests/test_domain_cpp.py [hunk 2]
```diff
@@ -1015,7 +1015,7 @@ def test_build_domain_cpp_semicolon(app, status, warning):
 
 @pytest.mark.sphinx(testroot='domain-cpp',
                     confoverrides={'nitpicky': True, 'strip_signature_backslash': True})
-def test_build_domain_cpp_backslash_ok(app, status, warning):
+def test_build_domain_cpp_backslash_ok2(app, status, warning):
```
Comment (tk0miya) [inline tests/test_domain_cpp.py]: @jakobandersen This is also.

File: tests/test_intl.py [hunk 3]
```diff
@@ -470,7 +470,7 @@ def test_text_table(app):
 @sphinx_intl
 @pytest.mark.sphinx('gettext')
 @pytest.mark.test_params(shared_result='test_intl_gettext')
-def test_gettext_toctree(app):
+def test_gettext_toctree2(app):
```
Comment (tk0miya) [inline tests/test_intl.py]: I'll check this is really needed later.

File: tests/test_intl.py [hunk 4]
```diff
@@ -482,7 +482,7 @@ def test_gettext_toctree(app):
 @sphinx_intl
 @pytest.mark.sphinx('text')
 @pytest.mark.test_params(shared_result='test_intl_basic')
-def test_text_toctree(app):
+def test_text_toctree2(app):
```
Comment (tk0miya) [inline tests/test_intl.py]: I'll check this is really needed later.

**Comment by tk0miya** [comment]
I'll merge this after all conflicts will be resolved.

File: tests/test_environment_indexentries.py [hunk 5]
```diff
@@ -69,8 +70,9 @@ def test_create_pair_index(app):
                                       ('ёлка', [('', '#index-5')]),
                                       ('Ель', [('', '#index-4')])],
                                      None])])
-    assert index[6] == ('Е', [('ёлка', [[], [('Sphinx', [('', '#index-5')])], None]),
-                               ('Ель', [[], [('Sphinx', [('', '#index-4')])], None])])
+    assert index[6] == ('Е',
+                        [('ёлка', [[], [('Sphinx', [('', '#index-5')])], None]),
+                         ('Ель', [[], [('Sphinx', [('', '#index-4')])], None])])
```
Comment (tk0miya) [inline tests/test_environment_indexentries.py]: I posted https://github.com/sphinx-doc/sphinx/pull/8430 as a fix of these indentation warning.

File: tests/test_domain_c.py (hunk 1 above)
Comment (jakobandersen) [reply tests/test_domain_c.py]: This one should keep the name ``test_union_definitions`` while the above should be renamed to ``test_struct_definitions``.

File: tests/test_domain_cpp.py (hunk 2 above)
Comment (jakobandersen) [reply tests/test_domain_cpp.py]: This one should probably be called ``test_build_domain_cpp_backslash_ok_true`` and the previous one ``test_build_domain_cpp_backslash_ok_false``, i.e., encoding the value of ``strip_signature_backslash`` in the name.

File: tests/test_domain_c.py (hunk 1 above)
Comment (tk0miya) [reply tests/test_domain_c.py]: Done in https://github.com/sphinx-doc/sphinx/pull/8433.

**Pull request merged**